sycl/test-e2e/format.py

```diff
@@ -98,6 +98,47 @@ def getMatchedFromList(self, features, alist):
         except ValueError as e:
             raise ValueError("Error in UNSUPPORTED list:\n%s" % str(e))
 
+    def make_default_features_list(self, expr, triple, add_default=True):
+        ### EXCEPTIONS LIST
+        # TODO: Define elsewhere?
+        exceptions = {}
+        exceptions["spir64"]={
+                "cuda":False, "hip":False, "hip_amd":False, "hip_nvidia":False,
+                "native_cpu":False,
+                }
+        exceptions["system"]={
+                "linux":True, "windows":False, "system-windows":False,
+                "run-mode":False, "TEMPORARY_DISABLED":False,
+                }
+        queried_features = []
+        for f in expr:
+            queried_features = queried_features + re.findall("[-+=._a-zA-Z0-9]+", f)
+
+        features = []
+        for f in queried_features:
+            if (exceptions[triple].get(
+                f,exceptions["system"].get(f,add_default))):
+                features.append(f)
+        return features
+
+    def select_triples_for_test(self, test):
+        # Check Triples
+        triples = set()
+        possible_triples = ["spir64"]
+        for triple in possible_triples:
+            unsupported=self.make_default_features_list(test.unsupported,triple,False)
```

Comment:
what is the `False` are here, can we add a var to describe it and pass that?

Reply:
this parameter for the function represents if by default we should add a feature to the list of features we have. We want to be as permissive as possible when building so any feature we see in an unsupported line is by default not added to the list of features, the opposite would be true for the features in the requires lines.

Reply:
gonna be honest i dont totally understand this explanation :)

Reply:
got it, thanks. if you could add a comment explaining this that would be really helpful